internal/pkg/cli/env_init.go

```diff
@@ -300,6 +326,93 @@ func (o *initEnvOpts) askCustomizedResources() error {
 	if o.NoCustomResources {
 		return nil
 	}
+	adjustOrImport, err := o.prompt.SelectOne(
+		envInitDefaultEnvConfirmPrompt, "",
+		envInitCustomizedEnvTypes)
+	if err != nil {
+		return fmt.Errorf("select adjusting or importing resources: %w", err)
+	}
+	switch adjustOrImport {
+	case envInitImportEnvResources.String():
+		return o.askImportResources()
+	case envInitAdjustEnvResources.String():
+		return o.askAdjustResources()
+	case envInitUseDefaultEnv.String():
+		return nil
+	}
+	return nil
+}
+
+func (o *initEnvOpts) askImportResources() error {
+	if o.ImportVPC.ID == "" {
+		vpcID, err := o.sel.VPC(envInitVPCSelectPrompt, "")
+		if err != nil {
+			if err == selector.ErrVPCNotFound {
+				log.Errorf(`No existing VPCs were found. You can either:
+- Create a new VPC first and then import it.
+- Use the default Copilot environment configuration.
+`)
+			}
+			return fmt.Errorf("select VPC: %w", err)
+		}
+		o.ImportVPC.ID = vpcID
+	}
+	if o.ImportVPC.PublicSubnetIDs == nil {
+		publicSubnets, err := o.sel.PublicSubnets(envInitPublicSubnetsSelectPrompt, "", o.ImportVPC.ID)
+		if err != nil {
+			if err == selector.ErrSubnetsNotFound {
+				log.Errorf(`No existing public subnets were found in VPC %s. You can either:
+- Create new public subnets and then import them.
+- Use the default Copilot environment configuration.`, o.ImportVPC.ID)
+			}
+			return fmt.Errorf("select public subnets: %w", err)
+		}
+		o.ImportVPC.PublicSubnetIDs = publicSubnets
+	}
+	if o.ImportVPC.PrivateSubnetIDs == nil {
+		privateSubnets, err := o.sel.PrivateSubnets(envInitPrivateSubnetsSelectPrompt, "", o.ImportVPC.ID)
+		if err != nil {
+			if err == selector.ErrSubnetsNotFound {
+				log.Errorf(`No existing private subnets were found in VPC %s. You can either:
+- Create new private subnets and then import them.
+- Use the default Copilot environment configuration.`, o.ImportVPC.ID)
+			}
+			return fmt.Errorf("select private subnets: %w", err)
+		}
+		o.ImportVPC.PrivateSubnetIDs = privateSubnets
+	}
+	return nil
+}
+
+func (o *initEnvOpts) askAdjustResources() error {
+	if o.AdjustVPC.CIDR.String() == emptyIPNet.String() {
+		vpcCIDRString, err := o.prompt.Get(envInitVPCCIDRPrompt, envInitVPCCIDRPromptHelp, validateCIDR,
+			prompt.WithDefaultInput(stack.DefaultVPCCIDR))
+		if err != nil {
+			return fmt.Errorf("get VPC CIDR: %w", err)
+		}
+		_, vpcCIDR, err := net.ParseCIDR(vpcCIDRString)
+		if err != nil {
+			return fmt.Errorf("parse VPC CIDR: %w", err)
```

Comment:
should we include the input in the error msg? or is that returned by the std library already?

Reply:
i believe the input is included

Reply:
Actually just carefully looked at it, this error ideally should never be triggered because we already have validation for the input. this is just because our `Get` method only return a string instead of IPNet method (we could make this change in the future), so that we need to parse the string to IPNet again...